include/hse/error.h

```diff
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2021 Micron Technology, Inc.  All rights reserved.
+ */
+
+/** @file hse.h
+ */
+
+#ifndef HSE_ERROR_H
+#define HSE_ERROR_H
+
+/* HSE error contexts */
+
+#define HSE_ECTX_NONE 0
```

Comment:
Should these be in an enum instead of macros? Enums are typically easier to work with from a language binding perspective.

Reply:
Either way is fine with me.

Reply:
Since we're exposing it to the user an enum should be more useful, so name it **enum hse_err_ctx**.
We should also take this opportunity to merge hse_err_t and merr_t to both be signed int64_t, then we can get rid of that horrid merr_to_hse_err() function.  Probably converting between signed to unsigned and back again is broken somewhere/somehow/someday...

Reply:
Why isn't this in the include/hse/types.h ???

Reply:
I can put it there if you feel that is the correct place for it.